src/main/java/org/prebid/server/auction/StoredResponseProcessor.java

```diff
@@ -98,6 +101,51 @@ private List<Imp> excludeStoredAuctionResponseImps(List<Imp> imps,
                 .collect(Collectors.toList());
     }
 
+    public List<AuctionParticipation> applyStoredBidResponseAdjustments(
+            List<AuctionParticipation> auctionParticipations) {
+
+        return auctionParticipations.stream()
+                .map(StoredResponseProcessor::applyStoredBidResponseAdjustments)
+                .collect(Collectors.toList());
+    }
+
+    private static AuctionParticipation applyStoredBidResponseAdjustments(AuctionParticipation auctionParticipation) {
+        final BidderRequest bidderRequest = auctionParticipation.getBidderRequest();
+        final BidRequest bidRequest = bidderRequest.getBidRequest();
+
+        final List<Imp> imps = bidRequest.getImp();
+        // for now, Stored Bid Response works only for bid requests with single imp
+        if (imps.size() > 1 || StringUtils.isEmpty(bidderRequest.getStoredResponse())) {
+            return auctionParticipation;
+        }
+
+        final BidderResponse bidderResponse = auctionParticipation.getBidderResponse();
+        final BidderSeatBid initialSeatBid = bidderResponse.getSeatBid();
+        final BidderSeatBid adjustedSeatBid = adjustSeatBid(initialSeatBid, imps.get(0).getId());
+
+        return auctionParticipation.with(bidderResponse.with(adjustedSeatBid));
+    }
+
+    private static BidderSeatBid adjustSeatBid(BidderSeatBid bidderSeatBid, String impId) {
+        final List<BidderBid> bids = bidderSeatBid.getBids().stream()
+                .map(bidderBid -> adjustBidImpId(bidderBid, impId))
+                .collect(Collectors.toList());
+
+        return bidderSeatBid.with(bids);
+    }
+
+    private static BidderBid adjustBidImpId(BidderBid bidderBid, String impId) {
```

Comment:
We use the 'adjust' word for another purpose in this project's scope.
Pls rename to `resolveBidImpId`

Reply:
Same for `adjustSeatBid` and `applyStoredBidResponseAdjustments` pls use another name e.g. `updateSeatBid`